srml/support/src/storage/hashed/generator.rs

```diff
@@ -205,6 +205,20 @@ pub trait StorageValue<T: codec::Codec> {
 		storage.put_raw(Self::key(), &new_val);
 		Ok(())
 	}
+
+	/// Read the length of the value in a fast way, without decoding the entire value.
+	///
+	/// `T` is required to implement `Codec::DecodeLength`.
+	fn len<S: HashedStorage<Twox128>>(storage: &mut S)
+		-> Result<usize, &'static str> where T: parity_codec_edge::DecodeLength
+	{
+		if let Some(k) = storage.get_raw(Self::key()) {
+			<T as parity_codec_edge::DecodeLength>::len(&k)
+				.ok_or_else(|| "Could not read len (value)")
```

Comment:
Maybe we should panic here directly.

Reply:
Absolutely not!

No user input (for these purposes, storage counts as user input) should ever lead to panicking.